drivers/macvlan/macvlan.go

```diff
@@ -51,17 +58,44 @@ type network struct {
 	endpoints endpointTable
 	driver    *driver
 	config    *configuration
+	once      *sync.Once
 	sync.Mutex
+	dbExists bool
+	dbIndex  uint64
 }
 
-// Init initializes and registers the libnetwork macvlan driver
+// Init registers a new instance of the macvlan driver
 func Init(dc driverapi.DriverCallback, config map[string]interface{}) error {
-	c := driverapi.Capability{
-		DataScope: datastore.LocalScope,
-	}
 	d := &driver{
 		networks: networkTable{},
 	}
+	// register the driver as globally scoped if a global scope label is passed
+	if labelData, ok := config[driverPrefix]; ok {
+		if labelData == globalScope {
```

Comment:
IMHO local scoped by default would seem to make sense for deploying Engine standalone and making things work out of the box.  I see global as the scope option when you're ready to do multi-host deployment.  But maybe I'm missing something?